### PR TITLE
Use java TimeUnit enum instead of raw multiplication for time unit conversions (#597)

### PR DESCRIPTION
**Pull request by 0x13a**
This refers to https://github.com/linkedin/ambry/issues/597

I've made a substitutions of `com.github.ambry.utils.Time` usages. It was used to convert **days -> Milliseconds** and vice versa via multiplication, instead of using [java.util.concurrent.TimeUnit](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html).

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java [hunk 1]
```diff
@@ -106,7 +107,7 @@
     this.diskIOScheduler = diskIOScheduler;
     this.time = time;
     scanSizeInBytes = Math.min(config.storeCleanupOperationsBytesPerSec * 10, 1024 * 1024);
-    messageRetentionSeconds = config.storeDeletedMessageRetentionDays * Time.SecsPerDay;
+    messageRetentionSeconds = Math.toIntExact(TimeUnit.DAYS.toSeconds(Integer.toUnsignedLong(config.storeDeletedMessageRetentionDays)));
```
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java]: wouldn't `messageRetentionSeconds = TimeUnit.DAYS.toSeconds(config.storeDeletedMessageRetentionDays)` suffice?

File: ambry-utils/src/main/java/com.github.ambry.utils/Time.java [hunk 2]
```diff
@@ -31,11 +31,6 @@
   public static final int NsPerSec = NsPerMs * MsPerSec;
   public static final int UsPerSec = UsPerMs * MsPerSec;
   public static final int SecsPerMin = 60;
```
Comment (vgkholla) [inline ambry-utils/src/main/java/com.github.ambry.utils/Time.java]: are the rest of these variables still in use?

Comment (0x13a) [inline ambry-utils/src/main/java/com.github.ambry.utils/Time.java]: SecsPerMin is used here https://github.com/linkedin/ambry/blob/master/ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java#L226

**Comment by 0x13a** [comment]
yes i will reintegrate master branch, no prob

File: ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java (hunk 1 above)
Comment (0x13a) [inline ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java]: the TimeUnit.DAYS.toSeconds https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html#toSeconds(long) accepts and returns long data type. Isn't that an issue ?

File: ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java [hunk 3]
```diff
@@ -22,13 +22,13 @@
 import com.github.ambry.utils.Time;
 import com.github.ambry.utils.Utils;
 import com.github.ambry.utils.UtilsTest;
-import java.util.ArrayList;
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java]: these removals seem to be causing build errors.
https://travis-ci.org/linkedin/ambry/builds/240916956

**Comment by vgkholla** [comment]
Changes look ok but it looks like you have build errors. Could you fix and run `./gradlew build` to verify?

**Comment by 0x13a** [comment]
alright, i will have to run it locally later ^^

**Comment by 0x13a** [comment]
I've probably fucked something up in the merge

File: ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
 import java.util.List;
 import java.util.NavigableMap;
 import java.util.Properties;
-import java.util.TreeMap;
```
Comment (vgkholla) [inline ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java]: just need to add this import back and I think you should be good to go.

**Comment by 0x13a** [comment]
Maybe next task would be working on code coverage

**Pull request merged**